shared/chat/conversation/input-area/normal/mention-input.js

```diff
@@ -152,7 +178,7 @@ class MentionInput extends React.Component<MentionInputProps, MentionState> {
     // This happens if you type @notausername<enter>. We've essentially 'picked' nothing and really want to submit
     // This is a little wonky cause this component doesn't directly know if the list is filtered all the way out
     if (options && options.notUser) {
-      this._forceSubmit()
+      this._submitAfterInsert()
```

Comment:
The comment here (and the similar case below) is slightly inaccurate, since it only mentions enter. I think what we want to do is:

1) make `_submitAfterInsert` take a boolean, say `notUserOrChannel`, and add that to the check.
2) rename it to `_maybeSubmitAfterInsert`
3) Combine the comment here (and below) with the comment in `_submitAfterInsert`, perhaps something like:

```
// If you type @notausername<enter> or #notachannel<enter>, we've essentially 'picked' nothing and really want to submit ... all the way out.
//
// If you type e.g., @notausername<tab> we don't want to do anything.
```